plugins_/syntaxtest_dev.py

```diff
@@ -375,19 +375,22 @@ def run(self, edit, character='^'):
 
 
 class AssignSyntaxTestSyntaxListener(sublime_plugin.EventListener):
-
     """Assign target syntax highlighting to a syntax test file."""
 
     def on_load_async(self, view):
+        plain_text = "Packages/Text/Plain text.tmLanguage"
         test_header = get_syntax_test_tokens(view)
         if test_header and test_header.syntax_file:
-            if view.settings().get('syntax', None) != test_header.syntax_file:
-                syntax = test_header.syntax_file
-                *_, file_name = syntax.rpartition('/')
-                if syntax in sublime.find_resources(file_name):
-                    view.assign_syntax(syntax)
-                else:  # file doesn't exist
-                    view.assign_syntax("Packages/Text/Plain text.tmLanguage")
+            view_syntax = view.settings().get('syntax', plain_text)
+            if view_syntax != test_header.syntax_file:
```

Comment:
The entire code block is fundamentally flawed, starting with this line. I'll just rewrite it.